torch/autograd/profiler.py

```diff
@@ -438,20 +438,17 @@ def __init__(self, name: str, args: Optional[str] = None):
         self.args: Optional[str] = args
         # Whether or not we should run record function's end callbacks when exiting.
         self.run_callbacks_on_exit: bool = True
-        # TODO: TorchScript ignores standard type annotation here
-        # self.record: Optional["torch.classes.profiler._RecordFunction"] = None
-        self.record = torch.jit.annotate(Optional["torch.classes.profiler._RecordFunction"], None)
+        # Stores underlying RecordFunction as a tensor. TODO: move to custom
+        # class (https://github.com/pytorch/pytorch/issues/35026).
+        self.handle: torch.Tensor = torch.zeros(1)
 
     def __enter__(self):
-        self.record = torch.ops.profiler._record_function_enter_new(self.name, self.args)
+        self.handle = torch.ops.profiler._record_function_enter(self.name, self.args)
```

Comment:
Would I be correct that this is intentionally using a legacy implementation because the new one is slower? If so, shall we add a N.B. comment to explicitly say that? Thanks!

https://github.com/pytorch/pytorch/blob/6ca8272d46a67b56b1cb65bb551bcbf5709f005b/torch/csrc/autograd/record_function_ops.cpp#L130